Replace Raster with Vector and Update Icons

theck13 opened the pull request:
### Fix
Replace raster (i.e. PNG) with vector (i.e. XML) icons and update deprecated icons with [Gridicons](https://github.com/Automattic/gridicons).  If a Gridicon didn't exist, official [Material icons](https://material.io/icons/) were used.  Some drawables (i.e. `ic_post_settings`) were left as PNG files intentionally since they are used as `drawableLeft` or `drawableRight` attributes.  Those attributes do not support vector drawables at the moment.  Unused drawables and their associated layouts were removed as well.

### Test
Navigate to as many screens as possible using the app as a user typical would.

#### Note
At a minimum, this pull request should be reviewed by a developer and a designer to ensure the changes follow the colors, iconography, and metrics laid out in the [WordPress.com Design Handbook](https://wordpress.com/design-handbook/).  Including one or more support people would be greatly appreciated also.

nbradbury commented:
I did a pretty thorough run through of the app and all looks good. I'm concerned, though, about hard-coding the colors in the vector files. Most of those colors are already defined in `colors.xml` (without the `#ff` alpha) so we should reference them using `@color`.

Also, I'm not sure I like including the hex color in the filename. That means if we change a color, we then change the names of (potentially) a lot of files. Would it be better to use more generic naming, such as `_light` and `_dark` and `_translucent`?

theck13 commented:
I agree; I would love to use color references in the vector files.  Will it work for pre-Lollipop devices?  [This issue](https://code.google.com/p/android/issues/detail?id=186431#hc13) says it won't, but I'm not sure.  I used to get a lint warning when replacing hardcoded colors with references in vector files, but I don't anymore.  Maybe it's been added to the support library?

Including the hexadecimal color in the filename was just a way to create some type of unambiguous naming convention.  If we can use the color references in the vector files, we could put those in the filenames (e.g. `ic_globe_blue_medium_24dp`).  My only concern with `_light` and `_dark` is the name is not always clear and multiple colors/icons could fall into each category.  For example, `ic_globe_ff000000_24dp` and `ic_globe_ff00aadc_24dp` could both be `ic_globe_light_24dp`.

nbradbury commented:
Wow, I wasn't aware of that limitation - that's kind of ridiculous. At any rate, it doesn't appear to be a problem now. I replaced a number of the fillColor values with `@color/alert_red` and everything appeared fine on an API 17 emulator.

![screenshot_1480509015](https://cloud.githubusercontent.com/assets/3903757/20752417/55a3b22c-b6cf-11e6-9e71-230c7949d37d.png)

Agreed about the naming convention. Using the named colors in the filenames would be much clearer for everyone.

theck13 commented:
This is the magic line in the `build.gradle` file which allows resource references in vector files.

```vectorDrawables.useSupportLibrary = true```

I added color references and renamed the vector files.  Let me know what you think.

nbradbury commented:
This all looks good to me! I know you said you'd like a designer to look at this so I'll wait to merge until you say it's ready.

theck13 commented:
> - We're not using the correct WordPress.com logotype.

I updated the logotype in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/c9646ea4550a8db5913080844456115cf756c0dd to use the version you provided.

> - Info icon should be white.

The info icon is currently `grey_lighten_10` so that it matches the `Add self-hosted site` and `Create account` buttons at the bottom of the screen.  Also, when the background changes from `blue_wordpress` to `f1f1f1` when you tap the `Add self-hosted site` button, the `grey_lighten_10` info button is much more visible than if it was white.  See the screenshots below for details.

![info](https://cloud.githubusercontent.com/assets/3827611/21171056/aba90778-c186-11e6-91d7-5ede470925b0.png)

> - Posts icon should be `grey_dark`, is `grey`.

I added a `grey_dark` version in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/83615b70e4866179b759a8043218593dd16e8233.

> - The time icon everywhere I've found it is out of date. Should use `gridicons-time`.

Unfortunately, the time icons fall under this caveat mentioned in the Fix section of the initial pull request comment.

> Some drawables (i.e. `ic_post_settings`) were left as PNG files intentionally since they ... do not support vector drawables at the moment.

> - Under Categories, X and + should use `gridicons-cross` and `gridicons-plus`
> - Under Location, pin icon should use `gridicons-location`

All three icons fall under the same caveat, i.e. they do not support vector drawables at the moment.

> - Plus sign in toolbar should use `gridicons-plus` instead of stock icon.

I updated the icon to `gridicons-plus` in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/d66282ded66237e08b151dc9aa55ccb978b2f295.

> - The pencil icon should use `gridicons-pencil` instead of stock icon.

I updated the icon to `gridicons-pencil` in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/ca492ccffc11ce6533b2d5c592ce44d978645d63.

> - To confirm, is the copy icon a material design icon?

Yes, it's from the [material design icon repository](https://material.io/icons/#ic_content_copy).

> In ••• menu, the View icon is currently 12px, others are 16px. All should be 18px instead. 

I updated the icons to 18dp in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/b2ddec9f784cc4880ef179c901e324d5c0fe3169.

> - Approved icon seems to be missing default padding, sitting out of alignment.

I added padding to the Approved and Like buttons in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/209c62fb00091c79260055634c6e6770a9e5365f.

> - Plus sign in toolbar should use gridicons-plus instead of stock icon.

I updated the icon to `gridicons-plus` in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/d66282ded66237e08b151dc9aa55ccb978b2f295.

> - Comment and Like icons are wrong size (should be 18px, currently 20px)

I updated the size to 18dp in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/fc5bcbaf6b187d2422cfea50339e15e818dbb996.  Note that the comment icon is not vertically centered intentionally; it's actually moved down 2dp to make the flat part of the comment align with the text baseline.  The screenshot below shows the layout bounds for that view.  See [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/4840#issuecomment-264657667) and [this commit](https://github.com/wordpress-mobile/WordPress-Android/pull/4840/commits/2b22256052b766311d305fbb7c274c5f0945bd5a) for details.

![comment](https://cloud.githubusercontent.com/assets/3827611/21204082/d13a7648-c212-11e6-9924-9e722a8e62a1.png)

> - Help & Support should use `gridicons-help`.

It's using `[gridicons-help](https://github.com/Automattic/gridicons/blob/master/svg/gridicons-help.svg)` from the Gridicons repository.  Are you suggesting it should use `[gridicons-help-outline](https://github.com/Automattic/gridicons/blob/master/svg/gridicons-help-outline.svg)`?

> - Icons are too large. See current version of app for correct sizing.

I updated the icons to 24dp in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/2d70138d56b3dd061f52ce1e91bebd80ef2534c1.

> - Small inset notification type icons (trophy, like, stats, etc) aren't updated to gridicons.

Those are actually Noticons which is determined from the API.  We should change that in another pull request.  I created a new issue [here](https://github.com/wordpress-mobile/WordPress-Android/issues/4921).

> - Icons are too big. Should be 24px.
> - Like and Approved icons have no padding, text labels are offset. 

This is the same view as Comment which I added padding to the Approved and Like buttons in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/209c62fb00091c79260055634c6e6770a9e5365f.  Should both views have 24dp icons?  They are currently 32dp.

> - Like icon should use `gridicons-star-outline` instead of `gridicons-star` when post/comment is not liked.

I updated the icon to `gridicons-star-outline` in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/2a92f0716193d9a04650bedfd1d227d7dc62ce2e.

> - "You replied to this comment" icon should be updated to `gridicons-reply`.

That is also a Noticon.  I created a new issue [here](https://github.com/wordpress-mobile/WordPress-Android/issues/4925).

mattmiklic commented:
> I updated the logotype in c9646ea to use the version you provided.

This looks good now! I see the WordPress.com logotype on the first screen now; is that intentional? On iOS, it only shows the W logo on the first screen, and expands to show the WordPress.com logotype when the password field appears. I think this is what we want to follow here; showing the WordPress.com logotype on the first screen could have some unintended side effects.

The W logo should match this: https://s.w.org/about/images/logos/wordpress-logo-simplified-cmyk.pdf -- note that the version of the W in Gridicons as `my-sites` has been adapted for uses only at 24px and below.

> Also, when the background changes from blue_wordpress to f1f1f1 when you tap the Add self-hosted site button, the grey_lighten_10 info button is much more visible than if it was white. See the screenshots below for details.

Ahh, I see. I think we should make that icon, and those text links, white or `blue_light` when on the blue login screen; the gray colors weren't really designed for text on a blue background. But that should probably be a separate issue.

> Note that the comment icon is not vertically centered intentionally; it's actually moved down 2dp to make the flat bottom part of the icon align with the text baseline. The screenshot below shows the layout bounds for that view. See this comment and this commit for details.

Gotcha, thanks for that link; I see the context now. Aligning the bottom of the comment icon to any text baseline will be difficult, because when the icon is displayed at 18px, the bottom of the comment bubble sits 0.25px off of the nearest whole pixel. So whether you move it up or down, it'll never align exactly. I think if you move it up by 1px, we'll get something that's as close to perfection as we can get, though. This is what 1px would look like:

<img width="894" alt="screen shot 2016-12-19 at 5 27 55 pm" src="https://cloud.githubusercontent.com/assets/1396552/21332023/c9c3f830-c613-11e6-8bd6-08ee1b29f0b8.png">

> It's using gridicons-help from the Gridicons repository. Are you suggesting it should use gridicons-help-outline?

Sorry, it looks like the PDF version of gridicons-help wasn't updated when the SVG version was, and I was looking at the PDF. This one is good to go.

> I updated the icons to 24dp in 2d70138.

The notification types icons are the right size now, but let's keep the same positioning as the current layout. I see you moved them to the 16dp keyline, which is a good idea, but I don't think it looks quite right for the way this layout has been designed. It might be worth looking at the layout again in a separate issue later on. Screenshot of current layout to match: https://cloudup.com/c5mRpqzvzMK

> Should both views have 24dp icons? They are currently 32dp.

I didn't catch that first time around in comments, but I see it now. Yes, I'd adjust those down to 24px. That's the size we use in notifications on mobile web.

Everything else is looking good!

theck13 commented:
> I see the WordPress.com logotype on the first screen now; is that intentional?

Yes.  That change was included in a separate [pull request](https://github.com/wordpress-mobile/WordPress-Android/pull/4851) and merged into this one to resolve conflicts.  The idea for adding the logotype was to avoid some confusion between the WordPress.com and WordPress.org logos on the login screen.  If we want to update the behavior, we can create a new issue.

> The W logo should match this: https://s.w.org/about/images/logos/wordpress-logo-simplified-cmyk.pdf

The WordPress.com logo in the logotype matches that.

> Ahh, I see. I think we should make that icon, and those text links, `white` or `blue_light` when on the blue login screen; the gray colors weren't really designed for text on a blue background. But that should probably be a separate issue.

I agree.  Can you create a new issue with specifications for the blue background screen (i.e. WordPress.com login) and the off-white background screen (i.e. WordPress.org login)?  Assign it to me and I'll code it up.

> I think if you move it up by 1px, we'll get something that's as close to perfection as we can get, though.

I moved the comment icon up 1dp in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/28293c78673405cb0e026c21da60e17d125c295c.  The result is shown below.

![comment](https://cloud.githubusercontent.com/assets/3827611/21335245/c1842904-c61a-11e6-89fe-4bff634a081c.png)

> The notification types icons are the right size now, but let's keep the same positioning as the current layout.

The old PNG's had some padding built into the image and the new vectors don't.  You can see the layouts below for comparison.

![notification](https://cloud.githubusercontent.com/assets/3827611/21335348/6380e1d4-c61b-11e6-9221-13475fc17c9e.png)

That screen is actually not a custom layout.  It's created by the operating system.  We supply the icon, title, and subtext.  Android does the rest.  We _can_ make a custom layout, but I think that should be a separate issue too.

> I didn't catch that first time around in comments, but I see it now. Yes, I'd adjust those down to 24px.

I updated the icons to 24dp in https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/1225bdf0655539b2c9cebe43c1a9930c4a5538fe.

maxme commented on WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:
```diff
@@ -112,6 +113,9 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main_activity);
 
+        // Allow vector drawables on pre-Lollipop (i.e. < API 21).
+        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
```
Note I added the same line in: https://github.com/wordpress-mobile/WordPress-Android/pull/4950

From the [doc](https://developer.android.com/reference/android/support/v7/app/AppCompatDelegate.html#setCompatVectorFromResourcesEnabled(boolean)):
> Please note: this only takes effect in Activities created after this call.

I think it's better to have this call in `WordPress.onCreate`, in case we use selectors in one of the tab fragment at some point or in the login screen.


From the [same doc](https://developer.android.com/reference/android/support/v7/app/AppCompatDelegate.html#setCompatVectorFromResourcesEnabled(boolean)), it's also a bit worrying:
> This feature defaults to disabled, since enabling it can cause issues with memory usage, and problems updating Configuration instances. If you update the configuration manually, then you probably do not want to enable this. You have been warned.

theck13 replied:
> If you update the configuration manually, then you probably do not want to enable this.

That's why I enabled it in `WPMainActivity.OnCreate` rather than `WordPress.OnCreate`.  We manually update the configuration using `onConfigurationChanged()` and `android:configChanges` in multiple activity and fragments throughout the app.

If you think it's better in `WordPress.OnCreate`, I can cherry pick your commit or just wait to merge with `develop`.

maxme replied:
`AppCompatDelegate.setCompatVectorFromResourcesEnabled` is a static call, so once it's set, it will be used everywhere and since `WPMainActivity` is created just after the app starts, this won't make a difference to have it here on in the app creation. 

Well the only difference: having it in the app onCreate(), allows us to have vector selectors in the sign in/sign up screens (called before `WPMainActivity`) or in the Tab fragment (created during `WPMainActivity` creation).

theck13 replied:
Sounds good.  I'll cherry pick your commit and remove it from `WPMainActivity.OnCreate`.

mattmiklic commented:
> Yes. That change was included in a separate pull request and merged into this one to resolve conflicts. The idea for adding the logotype was to avoid some confusion between the WordPress.com and WordPress.org logos on the login screen. If we want to update the behavior, we can create a new issue.

To clarify, the issue referred to in the PR you linked to says the goal is to match the implementation on iOS. On iOS, it looks like this:

<img width="1282" alt="screen shot 2016-12-22 at 9 58 11 am" src="https://cloud.githubusercontent.com/assets/1396552/21429554/539da81a-c82d-11e6-872e-3f5adc2a7738.png">

Note that on the first screen, only the "W" logo is shown. After tapping "Next," then the user sees the "WordPress.com" logotype. In this branch, the WordPress.com logotype is on the first screen. If the goal is to match the implementation on iOS, the W alone, without the WordPress.com text, should be shown on the first screen.

> I moved the comment icon up 1dp in 28293c7. The result is shown below.

Looks good now!

> The old PNG's had some padding built into the image and the new vectors don't. You can see the layouts below for comparison.

This is actually more problematic, I fear. Gridicons were designed with padding built in, too, it sounds like we've lost it somehow in the transition to Android. Gridicons need to include their padding, so that they are sized and aligned consistently. The layout issue I've described is a symptom of that padding not being present. I'm not sure what we want to do about that at this point, but I want to make sure we know that the icons should really include that padding, and we may have some design headaches in the future if they don't.

theck13 commented:
> This is actually more problematic, I fear. Gridicons were designed with padding built in, too, it sounds like we've lost it somehow in the transition to Android. Gridicons need to include their padding, so that they are sized and aligned consistently. The layout issue I've described is a symptom of that padding not being present. I'm not sure what we want to do about that at this point, but I want to make sure we know that the icons should really include that padding, and we may have more design headaches in the future if they don't.

The vectors in this pull request were all created from the SVG's in the [Gridicons](https://github.com/Automattic/gridicons) or [Material](https://material.io/icons/) repositories.  The viewbox/viewport and path values have not been modified when transitioning them to Android vector drawables.  The old PNG's in the case of the notification types list were given extra padding which I'm guessing was to make the icon horizontally centered between the screen edge and the text.  The new vectors icons don't have that extra padding; they have whatever padding was in the original Gridicon.

mattmiklic commented:
> The old PNG's in the case of the notification types list were given extra padding which I'm guessing was to make the icon horizontally centered between the screen edge and the text.

Ok, I think I understand this now. I'd still say these would look better if they could be horizontally centered between the text and the screen edge, but that's my only lingering piece of feedback.

theck13 commented:
> I'd still say these would look better if they could be horizontally centered between the text and the screen edge, but that's my only lingering piece of feedback.

I created a new issue [here](https://github.com/wordpress-mobile/WordPress-Android/issues/4955).

theck13 commented:
@mattmiklic, is the design review completed?  I believe I addressed or created issues for everything.

nbradbury commented:
@theck13  On an API 17 or API 19 emulator, I get this crash when tapping a reader-based notification or viewing reader detail. The problem doesn't occur on an API 23 emulator.
```
android.view.InflateException: Binary XML file line #105: Error inflating class <unknown>
    at android.view.LayoutInflater.createView(LayoutInflater.java:613)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
    at org.wordpress.android.ui.reader.ReaderPostDetailFragment.onCreateView(ReaderPostDetailFragment.java:194)
    at android.app.Fragment.performCreateView(Fragment.java:1699)
    at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:885)
    at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1057)
    at android.app.BackStackRecord.run(BackStackRecord.java:682)
    at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1435)
    at android.app.FragmentManagerImpl.executePendingTransactions(FragmentManager.java:474)
    at android.support.v13.app.FragmentStatePagerAdapter.finishUpdate(FragmentStatePagerAdapter.java:172)
    at android.support.v4.view.ViewPager.populate(ViewPager.java:1272)
    at android.support.v4.view.ViewPager.populate(ViewPager.java:1120)
    at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1646)
    at android.view.View.measure(View.java:15518)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:4825)
```

nbradbury commented:
The follow button in the reader in this branch is dark gray.

![screenshot_1485449455](https://cloud.githubusercontent.com/assets/3903757/22341250/193d231a-e3be-11e6-909b-48da3b46e731.png)

It should be blue, as it was changed in a recent reader redesign.

![screenshot_1485449704](https://cloud.githubusercontent.com/assets/3903757/22341299/54a2f376-e3be-11e6-9bcf-0a27700681e9.png)

nbradbury commented:
I received this crash after publishing a post from an API 19 emulator.
```
android.app.RemoteServiceException: Bad notification posted from package org.wordpress.android: 
Couldn't expand RemoteViews for: StatusBarNotification(pkg=org.wordpress.android user=UserHandle{0} id=1893 tag=null score=0: Notification(pri=0 contentView=org.wordpress.android/0x1090064 vibrate=null sound=null defaults=0x0 flags=0x10 kind=[null] 1 action))
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:136)
    at android.app.ActivityThread.main(ActivityThread.java:5017)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:515)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
    at dalvik.system.NativeStart.main(Native Method)
```

maxme commented:
> I received this crash after publishing a post from an API 19 emulator.

Similar to: https://github.com/wordpress-mobile/WordPress-Android/issues/4968#issuecomment-269999631

theck13 commented:
Odd.  I made a change (https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/59a4df273bafa3ba44038ed9f3527504a1f8716b) to use PNG's for notifications to avoid that specific crash.

theck13 commented:
> I get this crash when tapping a reader-based notification or viewing reader detail.

The [`reader_fragment_post_detail`](https://github.com/wordpress-mobile/WordPress-Android/pull/4807/files#diff-05864258876dd928af72ea5f49645bc3R113) layout was causing the crash.  Along with `drawableLeft` and `drawableRight`, `drawableTop` doesn't support vector drawables.  That should be fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/3e33b92bade20d2f53e84369e43da6925d6aa469.

> The follow button in the reader in this branch is dark gray.  It should be blue, as it was changed in a recent reader redesign.

That should be fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/3959cb632157234c1f750b07be014d1763ad3ad6.

nbradbury commented:
I'm still seeing a crash after publishing a post from an API 16 emulator.

```
com.android.systemui E/StatusBar: couldn't inflate view for notification org.wordpress.android/0x778
    android.content.res.Resources$NotFoundException: File res/drawable/ic_share_white_24dp.xml from drawable resource ID #0x7f0201b8
    at android.content.res.Resources.loadDrawable(Resources.java:1923)
    at android.content.res.Resources.getDrawable(Resources.java:659)
    at android.widget.TextView.setCompoundDrawablesWithIntrinsicBounds(TextView.java:1826)
    at android.widget.RemoteViews$TextViewDrawableAction.apply(RemoteViews.java:1181)
    at android.widget.RemoteViews.performApply(RemoteViews.java:2169)
    at android.widget.RemoteViews.apply(RemoteViews.java:2129)
    at android.widget.RemoteViews$ViewGroupAction.apply(RemoteViews.java:1113)
    at android.widget.RemoteViews.performApply(RemoteViews.java:2169)
    at android.widget.RemoteViews.apply(RemoteViews.java:2129)
    at com.android.systemui.statusbar.BaseStatusBar.inflateViews(BaseStatusBar.java:582)
    at com.android.systemui.statusbar.BaseStatusBar.addNotificationViews(BaseStatusBar.java:750)
    at com.android.systemui.statusbar.phone.PhoneStatusBar.addNotification(PhoneStatusBar.java:732)
    at com.android.systemui.statusbar.CommandQueue$H.handleMessage(CommandQueue.java:272)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:137)
    at android.app.ActivityThread.main(ActivityThread.java:4745)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:511)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
    at dalvik.system.NativeStart.main(Native Method)
    Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
    at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:877)
    at android.graphics.drawable.Drawable.createFromXml(Drawable.java:818)
    at android.content.res.Resources.loadDrawable(Resources.java:1920)
    at android.content.res.Resources.getDrawable(Resources.java:659) 
    at android.widget.TextView.setCompoundDrawablesWithIntrinsicBounds(TextView.java:1826) 
    at android.widget.RemoteViews$TextViewDrawableAction.apply(RemoteViews.java:1181) 
    at android.widget.RemoteViews.performApply(RemoteViews.java:2169) 
    at android.widget.RemoteViews.apply(RemoteViews.java:2129) 
    at android.widget.RemoteViews$ViewGroupAction.apply(RemoteViews.java:1113) 
    at android.widget.RemoteViews.performApply(RemoteViews.java:2169) 
    at android.widget.RemoteViews.apply(RemoteViews.java:2129) 
    at com.android.systemui.statusbar.BaseStatusBar.inflateViews(BaseStatusBar.java:582) 
    at com.android.systemui.statusbar.BaseStatusBar.addNotificationViews(BaseStatusBar.java:750) 
    at com.android.systemui.statusbar.phone.PhoneStatusBar.addNotification(PhoneStatusBar.java:732) 
    at com.android.systemui.statusbar.CommandQueue$H.handleMessage(CommandQueue.java:272) 
    at android.os.Handler.dispatchMessage(Handler.java:99) 
    at android.os.Looper.loop(Looper.java:137) 
    at android.app.ActivityThread.main(ActivityThread.java:4745) 
    at java.lang.reflect.Method.invokeNative(Native Method) 
    at java.lang.reflect.Method.invoke(Method.java:511) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
    at dalvik.system.NativeStart.main(Native Method) 
    03-07 17:46:56.616 4680-4680/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
    android.app.RemoteServiceException: Bad notification posted from package org.wordpress.android: Couldn't expand RemoteViews for: StatusBarNotification(pkg=org.wordpress.android id=1912 tag=null score=0 notn=Notification(pri=0 contentView=org.wordpress.android/0x1090071 vibrate=null sound=null defaults=0x0 flags=0x10 kind=[null] 1 action))
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1361)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:137)
    at android.app.ActivityThread.main(ActivityThread.java:4745)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:511)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
    at dalvik.system.NativeStart.main(Native Method)
```

theck13 commented:
> I'm still seeing a crash after publishing a post from an API 16 emulator.

That should be fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/5e9cbb11a7fd4d30285a8b12989d7e03257683c0.  I checked other notifications and they should be fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/4807/commits/3d356aee4e14e9bcd0097929154cc1ac57deb481.

nbradbury commented:
@theck13 Everything looks good! Once the 7.0 beta goes out we can :shipit:

nbradbury commented:
:shipit: 🎉
